src/pages/index.js

```diff
@@ -12,56 +11,204 @@ import {
   profileTitle,
   profileSubtitle,
   modalCard,
+  propicForm,
+  propicModal,
+  propicButtonEdit,
+  propic,
+  deleteModal,
+  deleteForm,
+  cardModalImage,
+  cardModalTitle,
 } from "../utils/constants.js";
-import { createCard, fillProfileForm } from "../utils/utils.js";
+import {
+  createCard,
+  fillProfileForm,
+  renderLoading,
+  checkResponse,
+} from "../utils/utils.js";
 import { UserInfo } from "../components/UserInfo.js";
 import FormValidator from "../components/FormValidator.js";
 import Section from "../components/Section.js";
 
 import { PopupWithForm } from "../components/PopupWithForm.js";
 import { PopupWithImage } from "../components/PopupWithImage.js";
 
-const cardsSection = new Section(
-  {
-    data: initialCards,
-    renderer: (item) => {
-      const cardElement = createCard(item);
-      cardsSection.addItem(cardElement);
-    },
-  },
-  cardList
-);
-cardsSection.renderItems();
+import { Api } from "../components/Api.js";
+import { PopupWithConfirmation } from "../components/PopupWithConfirmation.js";
 //
-export const cardPopup = new PopupWithImage(modalCard);
-cardPopup.setEventListeners();
+export const previewPopup = new PopupWithImage(
+  modalCard,
+  cardModalImage,
+  cardModalTitle
+);
+previewPopup.setEventListeners();
+
+const profileInfo = new UserInfo(profileTitle, profileSubtitle, propic);
+
+const profileFormValidator = new FormValidator(config, profileEditForm);
+profileFormValidator.enableValidation();
 
-const profileEddited = new UserInfo(profileTitle, profileSubtitle);
+const cardsFormValidator = new FormValidator(config, cardsEditForm);
+cardsFormValidator.enableValidation();
+
+const propicFormValidator = new FormValidator(config, propicForm);
+propicFormValidator.enableValidation();
 
 const profilePopup = new PopupWithForm(profileEditModal, {
   handleFormSubmit: (obj) => {
-    profileEddited.setUserInfo(obj);
+    api
+      .editProfile(obj)
+      .then((r) => {
+        profilePopup.close();
+        profileInfo.setUserInfo(r);
+      })
```

Comment:
You need to `catch` possible errors at the very **end** of any server request.

**Please, correct it everywhere**